docs/src/main/asciidoc/undertow-reference.adoc

```diff
@@ -29,3 +29,39 @@ jar files as defined by the Servlet spec.
 You can make use of the Undertow predicate language using an `undertow-handlers.conf` file. This file should be placed
 in the `META-INF` directory of your application jar. This file contains handlers defined using the
 link:http://undertow.io/undertow-docs/undertow-docs-2.0.0/index.html#predicates-attributes-and-handlers[Undertow predicate language].
+
+== CORS filter
+
+Quarkus comes with a CORS filter which can be enabled in the Quarkus configuration file, `application.properties`:
+
+[source, properties]
+----
+quarkus.http.cors=true
+----
+
+[NOTE]
+====
+The `application.properties` file must be located in the `src/main/resources` source folder. 
+====
+
+This filter can also be configured using the following properties:
+
+[cols="<m,<m,<2",options="header"]
+|===
+|Property Name|Default|Description
+|quarkus.http.cors.origins||The comma-separated list of origins allowed for CORS. The filter allows any origin if this is not set.
+|quarkus.http.cors.methods||The comma-separated list of HTTP methods allowed for CORS. The filter allows any method if this is not set.
+|quarkus.http.cors.headers||The comma-separated list of HTTP headers allowed for CORS. The filter allows any header if this is not set.
+|quarkus.http.cors.exposed-headers||The comma-separated list of HTTP headers exposed in CORS.
+|===
+
+Here's what a full CORS filter configuration could look like:
+
+[source, properties]
+----
+quarkus.http.cors=true
+quarkus.http.cors.origins=http://foo.com,http://www.bar.io
+quarkus.http.cors.methods=GET,PUT,POST
+quarkus.http.cors.headers=X-Custom
+quarkus.http.cors.exposed-headers=Content-Disposition
```

Comment:
Apparently, the Keycloak extension has a different CORS config. It's not documented in the Keycloak guide though.

But I think we should at least mention that, if using the Keycloak extension, the CORS configuration should be done with the Keycloak extension configuration (it's a bit unfortunate the config is totally different but it's a subject for another day).